packages/x-components/src/components/result/result-provider.vue

```diff
@@ -0,0 +1,192 @@
+<script lang="ts">
+  import Vue, { VNode, CreateElement } from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { Result, ResultVariant } from '@empathyco/x-types';
+  import { XProvide } from '../decorators/injection.decorators';
+  import {
+    RESULT_KEY,
+    SELECTED_VARIANTS_KEY,
+    SET_RESULT_VARIANT_KEY
+  } from '../decorators/injection.consts';
+
+  /**
+   * Component that exposes the result merged with its selected variant in the default slot.
+   *
+   * It receives the original result and keeps track of the selected variant.
+   *
+   * It provides the original result, the array containing the selected variants and a callback to
+   * set the selected variant to be used from a child.
+   *
+   * @public
+   */
+  @Component
+  export default class ResultProvider extends Vue {
+    /**
+     * The original result containing the variants.
+     *
+     * @public
+     */
+    @Prop({
+      required: true
+    })
+    @XProvide(RESULT_KEY)
+    public result!: Result;
+
+    /**
+     * Array to keep track of the selected variants of the result.
+     * Each position of the array is a nest level in the variants' hierarchy, so,
+     * the second position will contain a variant that is present inside the variant of the first
+     * position, and so on.
+     *
+     * @public
+     */
+    @XProvide(SELECTED_VARIANTS_KEY)
+    public selectedVariants: ResultVariant[] = [];
+
+    /**
+     * Selects a variant of the result.
+     * When called, it slices the array of selected variants to remove the selected child variants.
+     * Emits the {@link XEventsTypes.UserSelectedAResultVariant} when called.
+     *
+     * @param level - The nest level where the variant is placed inside the result.
+     * @param variant - The variant to set.
+     * @public
+     */
+    @XProvide(SET_RESULT_VARIANT_KEY)
+    setResultVariant(level: number, variant: ResultVariant): void {
+      if (this.selectedVariants[level] === variant) {
+        return;
+      }
+      this.selectedVariants = this.selectedVariants.slice(0, level);
+      this.$set(this.selectedVariants, level, variant);
```

Comment:
You can do it in one step keeping the reactivity
```suggestion
      this.selectedVariants.splice(level, Number.POSITIVE_INFINITY, variant);
```